### DISCUSSION TITLE
Parsing DNS config file on Linux

### DISCUSSION THREAD

**Discussion by dhra** [lycheeverse/lychee#850] [General] [answered]
This is a line in my `/etc/resolve.conf` file:

```
search --
```

After running lychee:

```
lychee https://example.com/blog/ 
```

```
Error: Failed to create request client

Caused by:
    0: Network error
    1: builder error: error reading DNS system conf: Error parsing resolv.conf: ProtoError { kind: Msg("Malformed label: --") }
    2: error reading DNS system conf: Error parsing resolv.conf: ProtoError { kind: Msg("Malformed label: --") }
```

**Comment by mre** [comment, marked as answer]
Hey @dhra,
`--` is not a valid search domain as far as I know. Can you try to remove that line from your `resolve.conf` and try again?
In any case this is not a problem with lychee, but with the DNS resolver.
We use reqwest with `trust-dns`, which allows for [better concurrency under high load](https://github.com/seanmonstar/reqwest/issues/296), a similar issue you describe was [reported before](https://github.com/seanmonstar/reqwest/issues/435) and now `getaddrinfo` is the DNS resolver in reqwest again by default.
But as I said, I don't think `search --` is valid to begin with.

**Comment by mre** [comment]
Converting this issue to a discussion, since it doesn't track any kind of planned work.
